cyberdrop_dl/crawlers/thisvid.py

```
@@ -0,0 +1,329 @@
+COMMON_VIDEOS_TITLE_SELECTOR = "div#list_videos_common_videos_list"
+NEXT_PAGE_SELECTOR = "li.pagination-next > a"
+VIDEOS_SELECTOR = "a.tumbpu"
+
+
+class ThisVidCrawler(Crawler):
+    primary_base_domain = URL("https://thisvid.com")
+
+    def __init__(self, manager: Manager) -> None:
+        super().__init__(manager, "thisvid", "ThisVid")
+        self.request_limiter = AsyncLimiter(3, 10)
+
+    """~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"""
+
+    @create_task_id
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        """Determines where to send the scrape item based on the url."""
+        if any(p in scrape_item.url.parts for p in ("categories", "tags")) or scrape_item.url.query_string:
+            return await self.search(scrape_item)
+        elif "members" in scrape_item.url.parts:
+            return await self.profile(scrape_item)
+        elif "videos" in scrape_item.url.parts:
+            return await self.video(scrape_item)
+        raise ValueError
+
+    @error_handling_wrapper
+    async def search(self, scrape_item: ScrapeItem) -> None:
+        async with self.request_limiter:
+            soup: BeautifulSoup = await self.client.get_soup(self.domain, scrape_item.url, origin=scrape_item)
+        title = ""
+        if not scrape_item.url.query_string:
+            if category_title := soup.select_one(COMMON_VIDEOS_TITLE_SELECTOR):
+                common_title = category_title.get_text(strip=True)
+                if common_title.startswith("New Videos Tagged"):
+                    common_title = common_title.split("Showing")[0].split("Tagged with")[1].strip()
+                    title = f"{common_title} [tag]"
+                else:
+                    common_title = category_title.get_text(strip=True).split("New Videos")[0].strip()
+                    title = f"{common_title} [category]"
```

Assume `category_title` always has a value, otherwise you will silently ignore errors.

If for some reason it doesn't, it will throw an exception when you try to use it, which is good in this case. It means the site made some changes and the crawler needs to be updated.

```suggestion
            
            category_title = soup.select_one(COMMON_VIDEOS_TITLE_SELECTOR)
            common_title = category_title.get_text(strip=True)
            if common_title.startswith("New Videos Tagged"):
                common_title = common_title.split("Showing")[0].split("Tagged with")[1].strip()
                title = f"{common_title} [tag]"
            else:
                common_title = common_title.split("New Videos")[0].strip()
                title = f"{common_title} [category]"
```